agent/engine/containermetadata/acquire_metadata.go

```diff
@@ -0,0 +1,188 @@
+// Copyright 2017 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License"). You may
+// not use this file except in compliance with the License. A copy of the
+// License is located at
+//
+//	http://aws.amazon.com/apache2.0/
+//
+// or in the "license" file accompanying this file. This file is distributed
+// on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+// express or implied. See the License for the specific language governing
+// permissions and limitations under the License.
+
+package containermetadata
```

Comment:
I kept it in the engine mostly because initially the only place my methods are used outside my package is in the engine package. But since I have made significant changes elsewhere requiring `containermetadata`, it is worthwhile moving it outside.